crates/bevy_debug_draw/src/lib.rs

```diff
@@ -0,0 +1,155 @@
+        load_internal_asset!(app, SHADER_HANDLE, "debuglines.wgsl", Shader::from_wgsl);
+
+        app.init_resource::<DebugDraw>()
+            .init_resource::<DebugDrawConfig>()
+            .add_system_to_stage(CoreStage::Last, update);
+
+        let Ok(render_app) = app.get_sub_app_mut(RenderApp) else { return; };
+
+        render_app.add_system_to_stage(RenderStage::Extract, extract);
+
+        #[cfg(feature = "bevy_sprite")]
+        {
+            use bevy_core_pipeline::core_2d::Transparent2d;
+            use pipeline_2d::*;
+
+            render_app
+                .add_render_command::<Transparent2d, DrawDebugLines>()
+                .init_resource::<DebugLinePipeline>()
+                .init_resource::<SpecializedMeshPipelines<DebugLinePipeline>>()
+                .add_system_to_stage(RenderStage::Queue, queue);
+        }
+
+        #[cfg(feature = "bevy_pbr")]
+        {
+            use bevy_core_pipeline::core_3d::Opaque3d;
+            use pipeline_3d::*;
+
+            render_app
+                .add_render_command::<Opaque3d, DrawDebugLines>()
+                .init_resource::<DebugLinePipeline>()
+                .init_resource::<SpecializedMeshPipelines<DebugLinePipeline>>()
+                .add_system_to_stage(RenderStage::Queue, queue);
+        }
+    }
+}
+
+#[derive(Resource, Clone, Copy)]
+pub struct DebugDrawConfig {
+    /// Whether debug drawing should be shown.
+    ///
+    /// Defaults to `true`.
+    pub enabled: bool,
+    /// Whether debug drawing should ignore depth and draw on top of everything else.
+    ///
+    /// This setting only affects 3D. In 2D, debug drawing is always on top.
+    ///
+    /// Defaults to `true`.
+    pub always_on_top: bool,
+}
+
+impl Default for DebugDrawConfig {
+    fn default() -> Self {
+        Self {
+            enabled: true,
+            always_on_top: true,
+        }
+    }
+}
+
+#[derive(Component)]
+struct DebugDrawMesh;
+
+fn update(
+    config: Res<DebugDrawConfig>,
+    mut debug_draw: ResMut<DebugDraw>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut commands: Commands,
+) {
+    if let Some(mesh) = debug_draw
+        .mesh_handle
+        .as_ref()
+        .and_then(|handle| meshes.get_mut(handle))
+    {
+        if config.enabled {
+            debug_draw.update_mesh(mesh);
+        } else {
+            debug_draw.clear();
+            mesh.remove_attribute(Mesh::ATTRIBUTE_POSITION);
+            mesh.remove_attribute(Mesh::ATTRIBUTE_COLOR);
+        }
+    } else if config.enabled {
+        let mut mesh = Mesh::new(PrimitiveTopology::LineList);
+        debug_draw.update_mesh(&mut mesh);
+        let mesh_handle = meshes.add(mesh);
+        commands.spawn((
+            SpatialBundle::VISIBLE_IDENTITY,
+            DebugDrawMesh,
+            #[cfg(feature = "bevy_pbr")]
+            (mesh_handle.clone_weak(), NotShadowCaster, NotShadowReceiver),
+            #[cfg(feature = "bevy_sprite")]
+            Mesh2dHandle(mesh_handle.clone_weak()),
```

Comment:
I think this is still missing a `NoFrustrumCulling` component, otherwise the debug lines will only be visible if the origin is in the field of view (due to #4294)